pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1284,8 +1284,27 @@ public void addSchema(Schema schema, boolean override, boolean force)
     }
   }
 
+  public void updateSchemaDateTime(TableConfig tableConfig, Schema schema)
+      throws SchemaNotFoundException, SchemaBackwardIncompatibleException, TableNotFoundException {
+    String schemaName = schema.getSchemaName();
+    LOGGER.info("Updating segment zookeeper time interval metadata: {}", schemaName);
+
+    String tableNameWithType = tableConfig.getTableName();
+    List<SegmentZKMetadata> segmentZKMetadataList = getSegmentsZKMetadata(tableNameWithType);
+    for (SegmentZKMetadata segmentZKMetadata : segmentZKMetadataList) {
+      int version = segmentZKMetadata.toZNRecord().getVersion();
+      updateSegmentInterval(tableConfig, segmentZKMetadata, schema);
+      updateZkMetadata(tableNameWithType, segmentZKMetadata, version);
+    }
+  }
+
   public void updateSchema(Schema schema, boolean reload)
       throws SchemaNotFoundException, SchemaBackwardIncompatibleException, TableNotFoundException {
+    updateSchema(schema, reload, false);
+  }
+
+  public void updateSchema(Schema schema, boolean reload, boolean force)
```

Comment:
Suggest reverting this unrelated change. We might not want force and reload at same time because we don't want to reload backward incompatible schema change